plugins/modules/discovery_intent.py

```diff
@@ -64,65 +72,143 @@
+                            - SHA uses Secure Hash Algorithm (SHA) as your authentication protocol.
+                            - MD5 uses Message Digest 5 (MD5) as your authentication protocol and is not recommended.
+                            - Not required for NOAUTHNOPRIV snmp_mode.
+                        type: str
+                        choices: [ 'SHA', 'MD5' ]
+                    privacy_type:
+                        description:
+                            - Privacy type/protocol of the SNMP v3 protocol to be used in AUTHPRIV SNMP mode
+                            - Not required for AUTHNOPRIV and NOAUTHNOPRIV snmp_mode.
+                        type: str
+                        choices: [ 'AES128', 'AES192', 'AES256' ]
+                    privacy_password:
+                        description:
+                            - Privacy password of the SNMP v3 protocol to be used in AUTHPRIV SNMP mode
+                            - Not required for AUTHNOPRIV and NOAUTHNOPRIV snmp_mode.
+                        type: str
+            net_conf_port:
+                description:
+                    - To be used when network contains IOS XE-based wireless controllers.
+                    - This is used for discovery and the enabling of wireless services on the controllers.
+                    - Requires valid SSH credentials to work.
+                    - Avoid standard ports like 22, 80, and 8080.
                 type: str
-            port:
-                description: The HTTP(S) port number, which is mandatory for using HTTP credentials.
-                type: int
-            secure:
-                description: This is a flag for HTTP(S). Its usage is not mandatory for HTTP credentials.
-                type: bool
-      http_write_credential:
-        description: HTTP write credentials for hosting a device
-        type: dict
-        suboptions:
-            username:
-                description: The username for HTTP(S) authentication, which is mandatory when using HTTP credentials.
-                type: str
-            password:
-                description: The password for HTTP(S) authentication. Mandatory for utilizing HTTP credentials.
-                type: str
-            port:
-                description: The HTTP(S) port number, which is mandatory for using HTTP credentials.
-                type: int
-            secure:
-                description: This is a flag for HTTP(S). Its usage is not mandatory for HTTP credentials.
-                type: bool
       ip_filter_list:
-        description: List of IP adddrsess that needs to get filtered out from the IP addresses added
+        description: List of IP adddrsess that needs to get filtered out from the IP addresses passed.
         type: list
         elements: str
       discovery_name:
-        description: Name of the discovery task
+        description: Name of the discovery taski
```

Comment:
what is taski?
Do we meant to say "The name assigned to the discovery task." ? 
or.
Name of the discovery task. It allows you to assign a descriptive name to your device discovery operation.